core/src/main/java/tc/oc/pgm/stats/PlayerStats.java

```diff
@@ -61,17 +60,22 @@ public class PlayerStats implements StatHolder {
   private Duration longestFlagHold = Duration.ZERO;
   private Instant longestFlagHoldCache;
 
+  // Variable-based statistics
+  private final Map<Variable<?>, Supplier<Double>> variableStats;
```

Comment:
i don't really like PlayerStats being polluted with this. The point of PlayerStats is to hold state that exists nowhere else: how many kills, deaths, etc the player has. These variables already have somewhere else to be, they shouldn't be here.

It should be a higher level of abstraction that implements `StatHolder` that can direct the calls either to this, or to the variable-based one for non "Base" stat types

Reply:
I think I understand where you are coming from on this. Is the recommendation that we create a new StatHolder that contains a references the formula, and have a reference to that stat holder here? Could be misunderstanding but I think I understand the general sentiment